core/src/main/java/hudson/model/Queue.java

```diff
@@ -728,7 +728,11 @@ public boolean cancel(Task p) {
     }
 
     private void updateSnapshot() {
-        snapshot = new Snapshot(waitingList, blockedProjects, buildables, pendings);
+        Snapshot revised = new Snapshot(waitingList, blockedProjects, buildables, pendings);
+        if (LOGGER.isLoggable(Level.FINEST)) {
+            LOGGER.log(Level.FINEST, "{0} → {1}; leftItems={2}", new Object[] {snapshot, revised, leftItems.asMap()});
```

Comment:
Generally handy to know when the state of the queue changes, for whatever reason.